processor/cumulativetodeltaprocessor/internal/tracking/value.go

```diff
@@ -20,4 +20,22 @@ type ValuePoint struct {
 	ObservedTimestamp pcommon.Timestamp
 	FloatValue        float64
 	IntValue          int64
+	HistogramValue    *HistogramPoint
```

Comment:
Does this need to be a pointer?

Reply:
It doesn't *need* to be, but since the field will be unused for most instances of this struct (e.g. all non-histogram metrics) it seemed slightly more memory efficient to use a pointer. One word of data for the pointer vs the 5(?) words for the size of a HistogramPoint.

Happy to discuss though.

Reply:
Is setting that value to nil have any overhead? My feeling is that it's the same as using the zero value for FloatValue/IntValue, whichever one isn't used.

As for separating the Value type - maybe as part of a larger refactoring of the whole processor. But the way it's architected right now, we have two entry points (Sum/Histogram) and then a lot of common functionality in the middle (identities, previous value store, etc.) and then a branch towards the end where we convert values. (And a second branch _within_ sums to handle float vs. int.)

Unless that's all refactored into two distinct code paths, splitting the Value type in two would feel like kicking the can down the road to me. Ultimately there needs to be some data somewhere to distinguish between sum and histogram values (in `State` or `MetricPoint`), and adding a third "case" to the Value type seems like the least intrusive way to me.

Reply:
Thanks for the well thought out response. I think you're right that this is the best solution.